arch/Kconfig

```diff
@@ -425,6 +425,12 @@ config CPU_HAS_TEE
 	  Execution Environment (e.g. when it has a security attribution
 	  unit).
 
+config CPU_HAS_DCLS
```

Comment:
I'd expect this commit to also include the code associated with CONFIG_CPU_HAS_DCLS

Reply:
> I was thinking CPU_HAS_DCLS could be a global symbol that describes "Dual-redundant Core Lock-step" feature for all architectures, like CPU_HAS_TEE and CPU_HAS_MPU.

The intention was to make this an architecture independent feature symbol, hence a separate commit from the one that adds all arch-specific changes.